### ISSUE TITLE
Response<double> get error when got int

### ISSUE THREAD

**Issue by a43mrk** [flutterchina/dio#691]
Hello i got an error that Dio cannot cast integer numbers to double, in this end-point its supposed to return only float numbers. but he returns to integers :( this code was generated by retrofit and the author said to post here, because he don't want to make any changes on the generator.
The error occurs the respose is assigned.

### New Issue Checklist

- [x ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |  all    |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |   13.0   |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |  3.x    |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |   Xcode 11   |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |   100%   |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |   x   |
| Demo project link              | e.g. link to a demo project that highlights the issue |   x   |

```
 getFooBarById(id) async {
   ArgumentError.checkNotNull(id, 'id');
   const _extra = <String, dynamic>{};
   final queryParameters = <String, dynamic>{};
   final _data = <String, dynamic>{};
   final Response<double> _result = await _dio.request('/foo/barById/$id',
       queryParameters: queryParameters,
       options: RequestOptions(
           method: 'GET',
           headers: <String, dynamic>{},
           extra: _extra,
           baseUrl: baseUrl),
       data: _data);
   final value = _result.data;
   return Future.value(value);
 }
```

> Exception has occurred.
> DioError (DioError [DioErrorType.DEFAULT]: type 'int' is not a subtype of type 'double'
> #0 DioMixin.assureResponse (package:dio/src/dio.dart:1138:9)
> #1 DioMixin._request. (package:dio/src/dio.dart:904:14)
> #2 _rootRunUnary (dart:async/zone.dart:1134:38)
> #3 _CustomZone.runUnary (dart:async/zone.dart:1031:19)
> #4 _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
> #5 Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
> #6 Future._propagateToListeners (dart:async/future_impl.dart:709:32)
> #7 Future._complete (dart:async/future_impl.dart:514:7)
> #8 _SyncCompleter.complete (dart:async/future_impl.dart:53:12)
> #9 Future.any. (dart:async/future.dart:459:45)
> #10 _rootRunUnary (dart:async/zone.dart:1134:38)
> #11 _CustomZone.runUnary (dart:async/zone.dart:1031:19)
> #12 _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
> #13 Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
> #14 Future._propagateToListeners (dart:async/future_impl.dart:709:32)
> #15 Future._completeWithValue (dart:async/future_impl.dart:524:5)
> #16 Future._asyncComplete. (dart:async/future_impl.dart:554:7)
> #17 _rootRun (dart:async/zone.dart:1126:13)
> #18 _CustomZone.run (dart:async/zone.dart:1023:19)
> #19 _CustomZone.runGuarded (dart:async/zone.dart:925:7)
> #20 _CustomZone.bindCallbackGuarded. (dart:async/zone.dart:965:23)
> #21 _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
> #22 _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)

**Comment by a43mrk** [comment]
hellow! is Dio alive?
